### PR TITLE
DAOS-722 logging: Remove key names from logging

### PR DESCRIPTION
**Pull request by dpquigl**
To avoid logging sensitive information key names are being removed from logging statements on the server side.

Signed-off-by: David Quigley <david.quigley@intel.com>
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/include/daos/common.h [hunk 1]
```diff
@@ -105,10 +105,8 @@ char *DP_UUID(const void *uuid);
 
 char *daos_key2str(daos_key_t *key);
```
Comment (ashleypittman) [inline src/include/daos/common.h:106]: This line should go as well.

File: src/common/debug.c [hunk 2]
```diff
@@ -242,37 +242,3 @@ DP_UUID(const void *uuid)
 #define DF_KEY_MAX		8
```
Comment (ashleypittman) [inline src/common/debug.c:242]: These two might be need to be included as well.  Sorry for the noise.

**Comment by johannlombardi** [comment]
Is that landed on master?

**Comment by dpquigl** [comment]
@johannlombardi This fix is only for 0.9. We intend to make this change in master dependent on the release/developer build feature that Jeff put in but for now we need to get it in 0.9 so we can close out pre_rc1 sdl tickets.

**Comment by jolivier23** [comment]
Failures are unrelated to change in logging and are due to existing issues.   I see no reason to block this patch for those issues.

**Pull request merged**